Stardrop/Utilities/External/NexusClient.cs

```diff
@@ -349,50 +376,75 @@ public async static Task<bool> ValidateKey(string? apiKey)
             {
                 Program.helper.Log($"Failed to get the download link for Nexus Mods: {ex}", Helper.Status.Alert);
             }
-            client.Dispose();
 
             return null;
         }
 
-        public async static Task<string?> DownloadFileAndGetPath(string uri, string fileName)
+        public async Task<NexusDownloadResult> DownloadFileAndGetPath(string uri, string fileName)
         {
-            // Create a throwaway client
-            HttpClient client = new HttpClient();
-            client.DefaultRequestHeaders.Add("Application-Name", "Stardrop");
-            client.DefaultRequestHeaders.Add("Application-Version", Program.ApplicationVersion);
-            client.DefaultRequestHeaders.Add("User-Agent", $"Stardrop/{Program.ApplicationVersion} {Environment.OSVersion}");
-
+            var requestUri = new Uri(uri);
+            var downloadCancellationSource = new CancellationTokenSource();
+            var requestMessage = new HttpRequestMessage(HttpMethod.Get, requestUri);
             try
             {
-                var stream = await client.GetStreamAsync(new Uri(uri));
-                using (var fileStream = new FileStream(Path.Combine(Pathing.GetNexusPath(), fileName), FileMode.CreateNew))
+
+                var response = await _client.SendAsync(requestMessage, HttpCompletionOption.ResponseHeadersRead, downloadCancellationSource.Token);
```

Comment:
`ResponseHeadersRead` gives us the chance to start watching the HTTP stream is comes down over the wire. That `downloadCancellationSource` eventually gets passed into the `ModDownloadViewModel`, and is what the Cancel button triggers.